### DISCUSSION TITLE
Add control to windows_expand_args parameter

### DISCUSSION THREAD

**Discussion by JakubRN** [fastapi/typer#1076] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer


def main(name: str):
    typer.echo(f"Hello {name}")

if __name__ == "__main__":

    app = typer.Typer()
    app.command()(main)
    app(windows_expand_args=False)

    # typer.run(main)
```

### Description

Currently it is impossible to globally disable `windows_expand_args` option implemented in click, the only solution I found is the code above. I found this feature quite questionable, since it can expand arbitrary string even if it was encapsulated in quotation marks or special characters were escaped. 

one common example is invoking app with `--name "local/path/*"` where my argument gets expanded to a list of paths containing all of the files.

Since this behaviour is completely unexpected and difficult to track down, I think it is at least worth mentioning in the documentation of typer, even better if it was possible to disable it.

### Operating System

Windows

### Typer Version

0.13.0

### Python Version

3.11.0

**Comment by bhirsz** [comment]
I agree, I was also catched off-guard by this. I simply had list[str] that was not a path, but * were expanded (to empty strings).

**Comment by ncoghlan** [comment]
After running into this myself (and first finding the older discussion at https://github.com/fastapi/typer/discussions/562), I've realised there's a concrete problem with not being able to set this at the `typer.Typer` level: testing based on `typer.testing.CliRunner` won't see the extra `windows_expand_args=False` parameter, so it isn't straightforward to check that the underlying `click` behaviour has been correctly disabled.
